app/services/platform-apps/api/modules/display.ts

```diff
@@ -67,40 +50,71 @@ export class DisplayModule extends Module {
   resize(ctx: IApiContext, displayId: string, size: IVec2) {
     const entry = this.getDisplayEntry(displayId);
 
-    entry.size = size;
-    this.updateDisplay(entry);
+    entry.options.size = size;
+    this.updateDisplay(entry, ctx.pageTransform.getValue());
   }
 
   @apiMethod()
   move(ctx: IApiContext, displayId: string, position: IVec2) {
     const entry = this.getDisplayEntry(displayId);
 
-    entry.position = position;
-    this.updateDisplay(entry);
+    entry.options.position = position;
+    this.updateDisplay(entry, ctx.pageTransform.getValue());
   }
 
   @apiMethod()
   destroy(ctx: IApiContext, displayId: string) {
     this.destroyDisplay(displayId);
   }
 
-  private updateDisplay(displayEntry: IDisplayEntry) {
-    if (displayEntry.webviewVisible) {
-      displayEntry.display.resize(displayEntry.size.x, displayEntry.size.y);
-      displayEntry.display.move(
-        displayEntry.webviewPosition.x + displayEntry.position.x,
-        displayEntry.webviewPosition.y + displayEntry.position.y,
-      );
-    } else {
-      displayEntry.display.resize(0, 0);
+  private updateDisplay(displayEntry: IDisplayEntry, transform: IBrowserViewTransform) {
+    // The container isn't mounted, so get rid of any active displays and stop
+    if (!transform.mounted) {
+      if (displayEntry.display) {
+        displayEntry.display.destroy();
+        displayEntry.display = null;
+      }
+
+      return;
     }
+
+    // The display is mounted to the wrong window, destroy it
+    if (
+      displayEntry.display &&
+      displayEntry.display.electronWindowId !== transform.electronWindowId
+    ) {
+      displayEntry.display.destroy();
+      displayEntry.display = null;
+    }
+
+    // There is no active display, so create one
+    if (!displayEntry.display) {
+      // Create display
+      displayEntry.display = new Display(displayEntry.displayId, {
+        electronWindowId: transform.electronWindowId,
+        slobsWindowId: transform.slobsWindowId,
+        paddingColor: displayEntry.options.paddingColor,
```

Comment:
Just wondering why do we need style-like props when we're apparently dealing with display/window mgmt code.

Reply:
The "display" is the OBS name for the thing that shows the video output.  It's aspect ratio locked, so it includes some padding/gutter to fit into whatever size box you give it.